src/preprocessing/point_in_poly/winding_number_jacobson.jl

```diff
@@ -0,0 +1,136 @@
+
+    return winding_number
+end
+
+"""
+    WindingNumberJacobson(; shape=nothing, winding_number_factor=sqrt(eps()),
+                          hierarchical_winding=false)
+Algorithm for inside-outside segmentation of a complex shape proposed by Jacobson et al. (2013).
+
+# Keywords
+- `shape`: Complex shape returned by [`load_shape`](@ref) and is only required when using
+           `hierarchical_winding=true`.
+- `hierarchical_winding`: If set to `true`, an optimised hierarchical approach will be used
+                          which gives a significant speedup.
+                          It is only supported for 3D shapes yet.
+- `winding_number_factor`: For leaky shapes, a factor of `0.4` will give a better inside-outside segmentation.
+"""
+struct WindingNumberJacobson{ELTYPE}
+    winding_number_factor :: ELTYPE
+    winding               :: Union{NaiveWinding, HierarchicalWinding}
+
+    function WindingNumberJacobson(; shape=nothing, winding_number_factor=sqrt(eps()),
+                                   hierarchical_winding=false)
+        if hierarchical_winding && shape isa Nothing
+            throw(ArgumentError("`shape` must be of type `Shapes` when using hierarchical winding"))
+        end
+
+        winding = hierarchical_winding ? HierarchicalWinding(shape) : NaiveWinding()
+
+        return new{typeof(winding_number_factor)}(winding_number_factor, winding)
+    end
+end
+
+function (point_in_poly::WindingNumberJacobson)(mesh::TriangleMesh{3}, points;
+                                                store_winding_number=false)
+    (; winding_number_factor, winding) = point_in_poly
+
+    inpoly = falses(size(points, 2))
+
+    winding_numbers = Float64[]
+    store_winding_number && (winding_numbers = resize!(winding_numbers, length(inpoly)))
+
+    @threaded points for query_point in axes(points, 2)
+        p = point_position(points, mesh, query_point)
+
+        winding_number = winding(mesh, p)
+
+        winding_number /= 4pi
+
+        store_winding_number && (winding_numbers[query_point] = winding_number)
+
+        # Relaxed restriction of `(winding_number != 0.0)`
+        if !(-winding_number_factor < winding_number < winding_number_factor)
+            inpoly[query_point] = true
+        end
+    end
+
+    return inpoly, winding_numbers
+end
+
+function (point_in_poly::WindingNumberJacobson)(mesh::Polygon{2}, points;
+                                                store_winding_number=false)
+    (; winding_number_factor) = point_in_poly
+    (; edge_vertices) = mesh
+
+    inpoly = falses(size(points, 2))
+
+    winding_numbers = Float64[]
+    store_winding_number && (winding_numbers = resize!(winding_numbers, length(inpoly)))
+
+    @threaded points for query_point in axes(points, 2)
+        p = point_position(points, mesh, query_point)
+
+        winding_number = sum(edge_vertices, init=0.0) do edge
+            a = edge[1] - p
+            b = edge[2] - p
+
+            return atan(det([a b]), (dot(a, b)))
```

Comment:
```suggestion
            return atan(det([a b]) / (dot(a, b)))
```
Can this also just be a regular arctan?

Reply:
see comment above